Mathlib/MeasureTheory/Function/L1Space.lean

```diff
@@ -1443,6 +1449,56 @@ theorem toL1_smul' (f : α → β) (hf : Integrable f μ) (k : 𝕜) :
 
 end Integrable
 
+section restrict
+
+variable {E : Type*} [NormedAddCommGroup E] {f : α → E}
+
+lemma HasFiniteIntegral.restrict (h : HasFiniteIntegral f μ) {s : Set α} :
+    HasFiniteIntegral f (μ.restrict s) := by
+  refine lt_of_le_of_lt ?_ h
+  convert lintegral_mono_set (μ := μ) (s := s) (t := univ) (f := fun x ↦ ↑‖f x‖₊) (subset_univ s)
+  exact Measure.restrict_univ.symm
+
+lemma Integrable.restrict (f_intble : Integrable f μ) {s : Set α} :
+    Integrable f (μ.restrict s) :=
+  ⟨f_intble.aestronglyMeasurable.restrict, f_intble.hasFiniteIntegral.restrict⟩
+
+lemma Integrable.measure_preimage_eq_measure_restrict_preimage_of_ae_compl_eq_zero
+    {E : Type*} [NormedAddCommGroup E] [MeasurableSpace E] [BorelSpace E]
+    {f : α → E} (f_intble : Integrable f μ) {s : Set α}
+    (hs : f =ᵐ[Measure.restrict μ sᶜ] 0) {t : Set E} (t_mble : MeasurableSet t) (ht : 0 ∉ t) :
+    (μ (f ⁻¹' t)) = ((μ.restrict s) (f ⁻¹' t)) := by
+  rw [Measure.restrict_apply₀]
+  · simp only [EventuallyEq, Filter.Eventually, Pi.zero_apply, Measure.ae,
+                MeasurableSet.compl_iff, Filter.mem_mk, mem_setOf_eq] at hs
+    rw [Measure.restrict_apply₀] at hs
+    · apply le_antisymm _ (measure_mono (inter_subset_left _ _))
+      apply (measure_mono (Eq.symm (inter_union_compl (f ⁻¹' t) s)).le).trans
+      apply (measure_union_le _ _).trans
+      have obs : μ ((f ⁻¹' t) ∩ sᶜ) = 0 := by
+        apply le_antisymm _ (zero_le _)
+        rw [← hs]
+        apply measure_mono
+        gcongr
+        intro x hx hfx
+        simp only [mem_preimage, mem_setOf_eq] at hx hfx
+        exact ht (hfx ▸ hx)
+      simp only [obs, add_zero, le_refl]
+    · exact NullMeasurableSet.of_null hs
+  · exact f_intble.restrict.aemeasurable.nullMeasurable t_mble
+
+lemma restrict_aeeq_zero_iff_ae_mem_eq_zero [MeasurableSpace E] [MeasurableSingletonClass E]
```

Comment:
Is there a preferred spelling?

[`AEFinStronglyMeasurable.exists_set_sigmaFinite`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.html#MeasureTheory.AEFinStronglyMeasurable.exists_set_sigmaFinite) uses the LHS spelling.

[`set_integral_eq_integral_of_ae_compl_eq_zero`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Integral/SetIntegral.html#MeasureTheory.set_integral_eq_integral_of_ae_compl_eq_zero) uses the RHS spelling.

I have added `integral_eq_integral_restrict` as well as this equivalence `restrict_aeeq_zero_iff_ae_mem_eq_zero` essentially only for the reason to glue between the different spellings in the above two results.

Reply:
I'd say `ae_restrict_eq_zero_iff_ae_eq_zero_of_mem`.

Reply:
Ok, I will change the naming as suggested.

But by preferred spelling, I meant whether the LHS or the RHS of this "iff" should be used as a hypothesis and conclusion of results. Currently, `[AEFinStronglyMeasurable.exists_set_sigmaFinite](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Function/StronglyMeasurable/Basic.html#MeasureTheory.AEFinStronglyMeasurable.exists_set_sigmaFinite)` has a conclusion of the LHS form, and `[set_integral_eq_integral_of_ae_compl_eq_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Integral/SetIntegral.html#MeasureTheory.set_integral_eq_integral_of_ae_compl_eq_zero)` has a hypothesis of the RHS form. This mismatch is why combining the two is not a one-liner.

Reply:
I realize it might have been clearer if I tagged the lemma type instead of the lemma name for this question. So the iff is:
```
f =ᵐ[Measure.restrict μ s] 0 ↔ ∀ᵐ x ∂μ, x ∈ s → f x = 0 
```

And the question is which of the two sides of the iff is or should be "normal form"? If there is no preference, then the users are more often responsible for translating between the spellings. If there is a clear preference, I can mark the present lemma as `@[simp]` (possibly reversing the direction).

Reply:
Don't mark this as `simp`, there is no clear direction which is better. I think the current direction is fine.